### PR TITLE
Products endpoint content type

### LINKED ISSUE #6491: ActionView::MissingTemplate in api/order_cycles#products
## Error in OpenFoodNetwork USA and Coopcircuits

**ActionView::MissingTemplate** in **api/order_cycles#products**
Missing template api/order_cycles/products, api/base/products with {:locale=>[:en_US, :en], :formats=>[:json, :js, :html, :text, :js, :css, :ics, :csv, :vcf, :png, :jpeg, :gif, :bmp, :tiff, :mpeg, :xml, :rss, :atom, :yaml, :multipart_form, :url_encoded_form, :json, :pdf, :zip], :variants=>[], :handlers=>[:erb, :builder, :raw, :ruby, :coffee, :haml]}. Searched in:
  * "/home/openfoodnetwork/apps/openfoodnetwork/releases-old/2020-12-08-184619/app/views"
  * "/home/openfoodnetwork/apps/openfoodnetwork/releases-old/2020-12-08-184619/app/views"

[View on Bugsnag](https://app.bugsnag.com/yaycode/openfoodnetwork-usa/errors/5fd0218d20374d001835d3aa?event_id=5fd0218d00659ee313d30000&i=gh&m=ci)

## Stacktrace

    app/controllers/api/base_controller.rb:76 - error_during_processing

[View full stacktrace](https://app.bugsnag.com/yaycode/openfoodnetwork-usa/errors/5fd0218d20374d001835d3aa?event_id=5fd0218d00659ee313d30000&i=gh&m=ci)

*Created by Admin OFF via Bugsnag*

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #6491

Ensures the Angular `OrderCycleResource` is using json format, which means the response is rendered with the correct `content_type`. Following on from discussions in https://github.com/openfoodfoundation/openfoodnetwork/issues/6577

#### Release notes

Updated Angular OrderCycleResource routes to use json format explicitly.

Changelog Category: Technical changes

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]

**Comment by Matt-Yorkley** [comment]
@andrewpbrett I've updated the PR a bit and added a regression test for the bug. It was failing, but it's fixed now.

**Review by andrewpbrett — APPROVED** [review]
Looks great! 💪

File: spec/controllers/api/order_cycles_controller_spec.rb [hunk 1]
```diff
@@ -148,6 +148,20 @@ module Api
           expect(product_ids).to include product3.id
         end
       end
+
+      context "when the order cycle is closed" do
+        before do
+          allow_any_instance_of(OrderCycle).to receive(:open?) { false }
```
Comment (sauloperez) [inline spec/controllers/api/order_cycles_controller_spec.rb:154]: See https://github.com/openfoodfoundation/openfoodnetwork/pull/6534#discussion_r553881544

**Review by sauloperez — CHANGES_REQUESTED** [review]

File: spec/controllers/api/order_cycles_controller_spec.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply spec/controllers/api/order_cycles_controller_spec.rb]: Done :ok_hand:

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
This looks like a no-test to me. So we could merge right after rebasing and getting a :green_apple: build.

**Pull request merged**